scripts/check-versions.py

```diff
@@ -0,0 +1,269 @@
+    for package in packages:
+        if not check_version(package, silent=True)[0]:
+            update_package(package)
+
+
+def update_package(package):
+    print_debug("Updating %s" % package)
+    print_header()
+    match, version, dist_version = check_version(package, silent=False, details=True)
+    if match:
+        print("Packet is already in newest version")
+        return
+    if dist_version == "":
+        print("Packet not found in dist file")
+        return
+
+    if version == "git":
+        print("Layer uses git version. Please update manually") 
+        return
+    elif LooseVersion(version) > LooseVersion(dist_version) and not "--downgrade" in sys.argv:
+        print("Layer version is newer than dist version, use --downgrade as last argument to downgrade")
+        return
+
+    try:
+        path = os.path.join(BASE_DIR, package)
+        recipes = []
+        for filename in os.listdir(path):
+            if filename.endswith(".bb"):
+                recipes.append(os.path.join(path, filename))
+
+        update_include = False 
+        rename_requests = []
+        for recipe in recipes:
+            with open(recipe) as recipe_file:
+                data = recipe_file.read()
+                
+            old_fn = os.path.join(recipe)
+            new_fn = os.path.join(recipe.replace(version, dist_version))
+            rename_requests.append([old_fn, new_fn])
+
+            if "SRC_URI[md5sum]" not in data and "SRC_URI[sha256sum]" not in data:
+                update_include = True 
+            else:
+                update_checksums_in_file(package, old_fn, dist_version)
+
+        if update_include:
+            inc_fn = os.path.join(BASE_DIR, package, package + ".inc")
+            update_checksums_in_file(package, inc_fn, dist_version)
+    except MoveRepoExcetion:
+        print_err("Repo moved %s" % package)
+    else:
+        for [old_fn, new_fn] in rename_requests:
+            print_debug("Rename %s" % recipe)
+            #print("old: %s" % old_fn)
+            #print("new: %s" % new_fn)
+            os.rename(old_fn, new_fn)
+    print_ok("Updated")
+
+        
+
+
+def print_header():
+    print("\033[1m\033[4m" + "package".ljust(35) +
+          "layer".ljust(13) + "distro".ljust(10) + '\033[0m')
+
+
+def print_list(details=False):
+    dist_raw = yaml.load(urllib.request.urlopen(DIST_FILE).read())
+    packages = [x for x in sorted(os.listdir(BASE_DIR)) if x not in EXCLUDE]
+    print_header()
+    for package in packages:
+        check_version(package, details=details)
+
+
+def print_mismatch(details=False):
+    dist_raw = yaml.load(urllib.request.urlopen(DIST_FILE).read())
+    packages = [x for x in sorted(os.listdir(BASE_DIR)) if x not in EXCLUDE]
+    print_header()
+    for package in packages:
+        if not check_version(package, silent=True)[0]:
+            match, version, dist_ver = check_version(package, details=details)
```

Comment:
unused variables? Also `check_version()` could probably accept `details=<"mismatch" | "all" | "none">` to avoid double calling.